src/panels/lovelace/editor/hui-edit-card.ts

```diff
@@ -69,25 +75,37 @@ export class HuiEditCard extends hassLocalizeLitMixin(LitElement) {
       _saving: {},
       _loading: {},
       _isToggleAvailable: {},
+      _isSavingAvailable: {},
     };
   }
 
+  private get _dialog(): PaperDialogElement {
+    return this.shadowRoot!.querySelector("paper-dialog")!;
+  }
+
+  private get _previewEl(): HuiCardPreview {
+    return this.shadowRoot!.querySelector("hui-card-preview")!;
+  }
+  public cardConfig?: LovelaceCardConfig;
+  public viewId?: string | number;
+  protected hass?: HomeAssistant;
+  private _cardId?: string;
+  private _configElement?: LovelaceCardEditor | null;
+  private _uiEditor?: boolean;
+  private _configValue?: ConfigValue;
+  private _configState?: string;
+  private _loading?: boolean;
+  private _isToggleAvailable?: boolean;
```

Comment:
Do we need to store this as an instance variable or can this be derived from other instance values?

Reply:
We should aim to store as little instance variables as possible, all values that can be derived from others, should not be stored. If used at multiple places consider adding a getter that computes it on the fly.

The main reason for this is single source of truth. What if `isToggleAvailable` goes out of sync with the values it was computed off, what happens then? By computing it on the fly we don't have this problem.